### DISCUSSION TITLE
Option or hotkey to paste to the parent node of selected node

### DISCUSSION THREAD

**Discussion by Crare** [godotengine/godot-proposals#6976] [Editor]
Currently copying selected node and pasting it will paste the new node under the selected node. Some other engines default to pasting it to the parent node. I wish there was a option to do so. This would make copying and pasting nodes faster when you don't want the pasted nodes to be nested children of the last selected node. Copying and pasting currently ends up in a nested loop of nodes if done repeatly.

It could be a option that can be turned on and off from settings somewhere or another hotkey like (ctrl + shift+ v) to paste to parent node of the selected node, so that pasted node is on the same hierarchy level than the selected node. If there is no parent node, then use root node or the default to selected node if all else fails.

how it currently works:
at start:
- parent node
  - selected node

after copying and pasting:
- parent node 
  - selected node
    - pasted node

how I would like it to work:
at start:
- parent node
  - selected node

after copying and pasting:
- parent node
  - selected node
  - pasted node

**Comment by Calinou** [comment]
I think adding <kbd>Ctrl + Shift + V</kbd> to paste as a sibling makes sense if it's not used for other purposes in the scene tree dock right now.

We could also have <kbd>Shift + Enter</kbd> to add a node as sibling in the Create New Node dialog, but this wasn't easy to implement last time I checked.

**Reply by Crare** [reply]
I think `ctrl + V + something` would be familiar to most users and easy to remember.

**Reply by Calinou** [reply]
- Done: https://github.com/godotengine/godot/pull/79467
